prog/mutation.go

```diff
@@ -320,7 +320,28 @@ func (t *IntType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []*Ca
 }
 
 func (t *FlagsType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []*Call, retry, preserve bool) {
-	return mutateInt(r, s, arg)
+	if !t.BitMask {
+		return mutateInt(r, s, arg)
+	}
+
+	a := arg.(*ConstArg)
+	switch {
+	case r.nOutOf(1, 10):
+		a.Val = 0
+		for _, val := range t.Vals {
+			if r.bin() {
+				a.Val |= val
+			}
+		}
+	case r.nOutOf(2, 10):
+		a.Val = 0
+	default:
+		for stop := false; !stop; stop = r.oneOf(3) {
+			a.Val ^= t.Vals[r.rand(len(t.Vals))]
```

Comment:
This uses xor to invert flags, which is good. But consider that flags have 10 possible values and we have 2 currently set. In such situation setting bits will have much higher probability (8/10), while resetting flags will have 2/10. Overall this algo will try to enable 50% of flags.
I am thinking that maybe it's better to have separate cases for setting flags and resetting flags. Then adding few more flags will have the same probability as removing few existing flags, which looks like a better mutation unit.